packages/api/src/external/carequality/document/shared.ts

```diff
@@ -87,3 +92,60 @@ const generateCQFHIRContent = (
 
   return [cqFHIRContent];
 };
+
+function generateCQFHIRContained(authorInstitution: string | undefined): Organization | undefined {
+  if (!authorInstitution) return;
+
+  const org = splitNameAndOid(authorInstitution);
+  if (!org) return generateOrganization(authorInstitution);
+  const organization = generateOrganization(org.name, org.oid);
+
+  return organization;
+}
+
+export function generateOrganization(name?: string, oid?: string) {
+  const organization: Organization = {
+    resourceType: "Organization",
+  };
+  if (name) organization.name = name;
+  if (oid) {
+    organization.identifier = [
+      {
+        value: oid,
+      },
+    ];
+  }
+  return organization;
+}
+
+function splitNameAndOid(
+  input: string
+): { name: string | undefined; oid: string | undefined } | undefined {
+  const regex = /^(.+?)\^+(.+)$/;
+  const match = input.match(regex);
+
+  if (match && match.length === 3) {
+    return { name: match[1], oid: match[2] };
+  }
+
+  return undefined;
+}
+
+export function formatDate(dateString: string | undefined): string | undefined {
```

Comment:
The string comes in a format that dayjs/Date doesn't recognize, right? Just to make sure we tested it.

Also, if we keep this around, this seems like a CQ-general thing, not just for docs, no?

Reply:
Yea, they don't recognize it, sadly.. :(

Agree!